analyzer/analyzer_test.go

```diff
@@ -0,0 +1,117 @@
+package analyzer
+
+import (
+	"context"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/aquasecurity/fanal/extractor"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockDockerExtractor struct {
+	saveLocalImage  func(ctx context.Context, imageName string) (io.Reader, error)
+	extractFromFile func(ctx context.Context, r io.Reader, filenames []string) (extractor.FileMap, error)
+	extract         func(ctx context.Context, imageName string, filenames []string) (extractor.FileMap, error)
+}
+
+func (mde mockDockerExtractor) Extract(ctx context.Context, imageName string, filenames []string) (extractor.FileMap, error) {
```

Comment:
I could but my 2c for not using it are: 

* I find tests using testify/mock not very readable. 
* It adds extra overhead when defining tests as the setup needs to do runtime casting for objects, which could fail if done wrong and cause panics.
* The lesser the dependencies, the better.

Reply:
I definitely agree the fewer dependencies are better. Actually, because of the reason, I've avoided to use `stretchr/testify/assert` and used only standard libraries such as `reflect.DeepEqual` for tests. But now, we already depend on `stretchr/testify/assert`. If we use `stretchr/testify/mock`, I think it's not so big deal.

As for readability, I also agree with you. `stretchr/testify/mock` is poor readability. But if we use our own original mock, it's too flexible and freedom. I think some kind of restriction is necessary for everyone to write a similar code. `testify/mock` can restrict our writing. Besides, it can easily separate variable definitions and assertions. `mock.On(input).Return(output)`

As mentioned earlier, I don't think `testify/mock` is the best one. If you know other better ways, we can select it.